### ISSUE TITLE
Creating shaka instance causing 'loadstart' event to be raised

### ISSUE THREAD

**Issue by dan-ziv** [google/shaka-player#961]
Hi all,
On my player I have a logic that on 'loadstart' html5 event my player move from 'idle' state to 'loading' state and therefor showing a spinner.

When I want to start the player without preloading it, I'm only creating shaka instance:
`this._shaka = new shaka.Player(videoElement);`
`this._shaka.configure(config);`
Without loading it.

For some reason, the 'loadstart' html5 event is raised immediately after that.
Until we load() the player explicitly or click play, 'loadedmetada' will not raised and spinner will be shown ('loadedmetada' move the player to 'loaded' state and hides the spinner).

I checked your demo with the default manifest and the behavior is the same.
(You can place a break point in line 157, main.js:
`var localPlayer = new shaka.Player(localVideo);`
and write in the console:
`localVideo.onloadstart=function(){debugger;};`
and see it stopping on the debugger statement.

I'm using shaka version
v2.1.6

Thanks for your help!

**Comment by joeyparrish** [comment]
This could be caused by the fact that `Player` starts initializing `MediaSource` as soon as it is created.  That way, `load()` can have a lower latency.

In Shaka Player, I believe the best event to show/hide the loading spinner is the `'buffering'` event dispatched by `Player`.  For example:

```js
player.addEventListener('buffering', function(event) {
  spinner_.style.display = event.buffering ? 'block' : 'none';
});
```

Does this help?

**Comment by joeyparrish** [comment]
@dan-ziv, does the buffering event help at all?  Is there anything else we can do for you?

**Comment by dan-ziv** [comment]
Hi @joeyparrish 
Unfortunately no.
I'm rely on html5 events since I want to be able to use different libraries with different MSEs.
The problem is indeed that you are attaching the MSE to the video element when constructing shaka player.
What if I want to start fetching segments and just later attach the MSE to the video element? this is not possible?
hls.js for example, supplying to different methods for those actions: 
```
var hls = new Hls();
hls.attachMedia(video);
```

**Comment by OrenMe** [comment]
@joeyparrish I think that https://github.com/google/shaka-player/issues/880 might allow this kind of handling, no?
I think that the hls.js way is pretty convenient too. 
maybe for backward compatibility you should call "attachMedia" method by default, but if a config flag is passed to allow the preload then the user would have to call "attachMedia".
I would generalise it more, cause if a user wants finer control aabd not necessarily pre-load, then I would allow for user to just set in config option "I would attach it by myself when I need too" and then the user will need to call "attachMedia".
WDYT?

**Comment by joeyparrish** [comment]
Yes, as part of #880, we will have to change the API so that a video element is not required in the Player constructor.  MediaSource would be set up later.  Examples:

```js
// current flow
player = new shaka.Player(video);  // attach right away, set up MediaSource
player.load('foo.mpd');
```

```js
// preload flow
player = new shaka.Player();
player.preload('foo.mpd');  // start loading content
player.attach(video);  // now set up MediaSource and complete the pipeline
```

```js
// wait without preloading
player = new shaka.Player();
// time passes...
player.attach(video);  // MediaSource setup happens now, when you want it to
```

Instead of a config option as @OrenMe suggests, the argument to the constructor is optional.  This gives us API backward compatibility, too.

@dan-ziv, what you want is not possible in any existing release, but would be possible after #880 is solved.  So I'm going to tentatively close this as a duplicate of #880 and tag you there.  Please let me know if you think this is the wrong move, and we can reopen.
